src/main/org/ovirt/jenkins/plugin/OvirtCloud.java

```diff
@@ -0,0 +1,81 @@
+package main.org.ovirt.jenkins.plugin;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import hudson.model.Label;
+import hudson.slaves.Cloud;
+import hudson.slaves.NodeProvisioner;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+
+public class OvirtCloud extends Cloud {
+
+    private static final Logger LOGGER = Logger.getLogger(OvirtCloud.class.getName());
+
+    private Label label;
+
+    public Label getLabel() { return this.label; }
+
+    @DataBoundConstructor
+    public OvirtCloud(String name) {
+        super(name);
+    }
+
+    public boolean canProvision(Label label) { return this.label == label; }
+
+    public Collection<NodeProvisioner.PlannedNode> provision(final Label label, final int excessWorkload) {
+
+        LOGGER.log(Level.INFO, "Provisioning {0} VMs", excessWorkload);
+
+        List<NodeProvisioner.PlannedNode> provisionedList = new ArrayList<>();
+
+        for (int i = 0; i < excessWorkload; ++i) {
+            try {
+                oVirtProvision(label);
+            } catch (Exception e) {
+                // TODO: Do something here?
+            }
+        }
+
+
+        return provisionedList;
+    }
+
+
+    /**
+     * Provision an oVirt VM to connect back to Jenkins
+     *
+     * @param label
+     * @return true if successfully provisioned oVirt VM
+     * @throws Exception if something went wrong
+     */
+    private boolean oVirtProvision(Label label) throws Exception {
+        String virtualMachineLabel = resolveVirtualMachineLabel(label);
+
+        try {
+            // TODO: Provision a VM using oVirt SDK
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, "Could not provision oVirt VM: {0}", e.getMessage());
+            throw e;
+        }
+
+        return true;
+    }
+
+    /**
+     * Given Jenkins agent label as requested by user, return the matching oVirt VM label
+     *
+     * @param label
+     * @return oVirt VM label matching the given Jenkins agent label
+     */
+    private String resolveVirtualMachineLabel(Label label) {
+        // TODO: resolve matching oVirt VM label for the given Jenkins agent label
+        // TODO: Do we want to require users to configure VM labels with the same label or include some logic here?
```

Comment:
Not sure what is "resolving ovirt vm label for jenkins label. Do we need this functionality now?